rust/azure_iot_operations_connector/src/base_connector/managed_azure_device_registry.rs

```diff
@@ -408,45 +398,142 @@ where
+                                azure_device_registry::ErrorKind::ServiceError(_) => {
+                                    RetryError::permanent(e)
+                                }
+                                _ => {
+                                    // InvalidRequestArgument shouldn't be possible since timeout is already validated
+                                    // ValidationError shouldn't be possible since we shouldn't receive a notification with an empty asset name
+                                    // ObservationError, DuplicateObserve, and ShutdownError aren't possible for this fn to return
+                                    unreachable!()
+                                }
+                            }
+                        })
+                },
+            )
+            .await
+            {
+                Ok(asset) => AssetClient::new(
+                    asset,
+                    asset_ref,
+                    self.device_specification.clone(),
+                    self.connector_context.clone(),
+                ),
+                Err(e) => {
+                    log::error!("Failed to get Asset definition after retries: {e}");
+                    log::error!("Dropping asset create notification: {asset_ref:?}");
+                    // unobserve as cleanup
+                    let _ = Retry::spawn(
+                        RETRY_STRATEGY,
+                        async || -> Result<(), RetryError<azure_device_registry::Error>> {
+                            self.connector_context
+                                .azure_device_registry_client
+                                .unobserve_asset_update_notifications(
+                                    asset_ref.device_name.clone(),
+                                    asset_ref.inbound_endpoint_name.clone(),
+                                    asset_ref.name.clone(),
+                                    self.connector_context.default_timeout,
+                                )
+                                // retry on network errors, otherwise don't retry on config/dev errors
+                                .await
+                                .map_err(observe_error_into_retry_error)
+                        },
+                    )
+                    .await
+                    .inspect_err(|e| {
+                        log::error!(
+                            "Failed to unobserve asset update notifications after retries: {e}"
+                        );
+                    });
+                    continue;
+                }
+            };
+            return Some((
+                asset_client,
+                asset_client_update_observation,
+                asset_deletion_token,
+            ));
+        }
     }
 }
 
 /// An Observation for asset update events that uses
 /// multiple underlying clients to get full information for a
 /// [`ProtocolTranslator`] to use.
```

Comment:
```suggestion
/// [`ProtocolTranslator`] to use.
```

Unrelated to this PR but since we might not have a trait, would this generate incorrectly in the docs?